optimas/generators/ax/developer/multitask.py

```diff
@@ -274,11 +296,21 @@ def suggest(self, num_points: Optional[int]) -> List[dict]:
                     var.name: arm.parameters.get(var.name)
                     for var in self._varying_parameters
                 }
-                # SH for VOCS standard these will need to be 'variables'
-                # For now much match the trial parameter names.
-                point["ax_trial_id"] = trial_index
-                point["arm_name"] = arm.name
-                point["trial_type"] = trial_type
+                # SH We can use a discrete var here in vocs (converted for now to trial parameters)
+                # But unlike varying parameters the name refers to a fixed generator concept.
+                for trial_param in self._custom_trial_parameters:
+                    if trial_param.name == "trial_type":
+                        point[trial_param.name] = trial_type
+
+                # Generate unique _id and store mapping
+                current_id = self._next_id
+                self._id_mapping[current_id] = {
+                    "arm_name": arm.name,
+                    "ax_trial_id": trial_index,
+                    "trial_type": trial_type,
```

Comment:
I got rid of `ax_trial_id` and `arm_name` as parameters and just held them internally mapped to a single id. We discussed doing this before. But I don't think this works, as we may choose hi-fidelity points from a batch incorporated from previous history. And with this change, we lose that information. I would need to know those historical points.

But this would not fit what vocs currently supports. ax_trial_id would be open ended integer variable, and arm is stored as a string, again open ended (not a discrete set). Maybe the could be observables, but they are created in the gen. They really just need to be known.

Reply:
Alt. could be for historical points, to find best points from the whole set and issue new arm_id.

Reply:
This is reason for:
`FAILED tests/test_ax_generators.py::test_ax_multitask_with_history `

Reply:
So trial_type (hi/low fidelity) is now declared as a discrete variable in vocs. This is a vocs variable as it is given to each simulation. As string types are not supported in the Optimus internal `VaryingParameter`s structure, it converts to `TrialParameter` internally.

`ax_trial_id` (batch id) and `arm_name` (point identifier) are set up in the gen and passed back and forth as generator specific values (not part of vocs). I have continued to use `TrialParameters` so Optimas stores them in the history array and they can be read back in from previous histories. However, this does mean an Optimas format is being used in the generator.